tests/unit/mocks/array_mock.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+# (c) 2023, Denys Denysyev (ddenysyev@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING.GPLv3 or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+class MockArray:
+    def __init__(self, params):
+        self.name = params["name"]
+
+        if not params["display_name"]:
+            self.display_name = params["name"]
+        else:
+            self.display_name = params["display_name"]
+
+        self.host_name = params["host_name"]
+        self.region = params["region"]
+        self.availability_zone = params["availability_zone"]
+        self.hardware_type = params["hardware_type"]
+        self.appliance_id = params["appliance_id"]
+
+        if not params["maintenance_mode"]:
+            self.maintenance_mode = True
+        else:
+            self.maintenance_mode = params["maintenance_mode"]
```

Comment:
@DenDen039 Sorry for pointing out so late, but now I remembered you can also use this
```
self.maintenance_mode = params.get("maintenance_mode", True)
```
It's up to you whether you want to add it or not... but it might come handy in the future.